server/api/utils/singletons/k8s.py

```diff
@@ -46,6 +46,21 @@ def get_k8s_helper(namespace=None, silent=True, log=False) -> "K8sHelper":
     return _k8s
 
 
+def raise_for_status_code(func):
+    """
+    A decorator for calls to k8s api when no error handling is needed.
+    Raises the matching mlrun exception to the status code.
+    """
+
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except ApiException as exc:
+            mlrun.errors.raise_for_status_code(exc.status, message=exc.reason)
```

Comment:
you lose traceback this one, try making it maybe
```suggestion
            raise mlrun.errors.raise_for_status_code(exc.status, message=exc.reason) from exc
```

and raise_for_status_code will simply return the exception (not raising it)